smac/utils/sklearn_wrapper.py

```diff
@@ -0,0 +1,324 @@
+
+        verbose : integer
+            Controls the verbosity: the higher, the more messages.
+
+        random_state : int or RandomState
+            Pseudo random number generator state used for random uniform sampling
+            from lists of possible values instead of scipy.stats distributions.
+
+        error_score : 'raise' (default) or numeric
+            Value to assign to the score if an error occurs in estimator fitting.
+            If set to 'raise', the error is raised. If a numeric value is given,
+            FitFailedWarning is raised. This parameter does not affect the refit
+            step, which will always raise the error.
+
+        return_train_score : boolean, default=True
+            If ``'False'``, the ``cv_results_`` attribute will not include training
+            scores.
+            """
+
+        # Current implementation does not like distributions yet.
+        for param, values in param_distributions.items():
+            if not isinstance(values, list):
+                raise ValueError('Not implemented: Wrapper does not work with distributions yet. Please use list. ')
+
+        self.random_state = random_state
+        self.param_distributions = param_distributions
+        self.n_iter = n_iter
+
+        # sets important hyperparameters of search module
+        super(ModelBasedOptimization, self).__init__(
+              estimator=estimator, scoring=scoring, fit_params=fit_params,
+              n_jobs=1, iid=iid, refit=refit, cv=cv, verbose=verbose,
+              pre_dispatch=False, error_score=error_score,
+              return_train_score=return_train_score)
+
+    def fit(self, X, y=None, groups=None):
+        """Run fit with all sets of parameters.
+
+        Parameters
+        ----------
+
+        X : array-like, shape = [n_samples, n_features]
+            Training vector, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples] or [n_samples, n_output], optional
+            Target relative to X for classification or regression;
+            None for unsupervised learning.
+
+        groups : array-like, with shape (n_samples,), optional
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+        """
+
+        # the following part is directly copied from Scikit-learns RandomizedSearchCV class
+        estimator = self.estimator
+        cv = check_cv(self.cv, y, classifier=is_classifier(estimator))
+        self.scorer_ = check_scoring(self.estimator, scoring=self.scoring)
+
+        X, y, groups = indexable(X, y, groups)
+        n_splits = cv.get_n_splits(X, y, groups)
+
+        base_estimator = clone(self.estimator)
+
+        cv_iter = list(cv.split(X, y, groups))
+
+        #################################### START DIFFERENCE WITH BASESEARCH_CV ####################################
```

Comment:
If I could, I obviously would have done it. It's a long sklearn function, and we need mostly the same functionality, except for the few marked lines. 

Do you have any suggestion how to avoid this?